Mamba 2.0 SPM Support

rmigneco opened the pull request:
### Description

This PR adds support for Swift Package Manager (SPM)

### Change Notes

* Adds `Package.swift` file
* Updates Project Dir Structure to support multiple targets for SPM to manage C and Swift sources with interdependencies
* Updates Xcode project file to reflect moved files and ensures Framework builds and tests run
* Fix import statements where necessary

### Outstanding items

- [x] Handle project version for SPM (possibly a `version.txt` file as a resource)
- [x] Ensure CocoaPods integration works
- [x] Ensure Carthage Integration works
- [x] Update README

### Pre-submission Checklist

- [x] I ran the unit tests locally before checking in.
- [x] I made sure there were no compiler warnings before checking in.
- [x] I have written useful documentation for all public code.
- [x] I have written unit tests for this new feature.

### Attachments
- example project demonstrating SPM integration
- example project demonstrating Carthage integration
- 
[MambaCarthageTest.zip](https://github.com/user-attachments/files/17688834/MambaCarthageTest.zip)
[MambaSPMTest.zip](https://github.com/user-attachments/files/17688835/MambaSPMTest.zip)

rmigneco committed:
Added package.swift

- separate targets for swift and objectiveC code

rmigneco committed:
Directory re-organization to support different targets

rmigneco committed:
Fix Xcode project file references

rmigneco commented:
@theRealRobG Confirmed installation with Carthage and SPM both work with attached demo projects. It's been a while since I've done CocaPods so might need an assist there.

theRealRobG commented on mamba.xcodeproj/project.pbxproj:
```diff
@@ -2376,6 +2460,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				BUILD_LIBRARY_FOR_DISTRIBUTION = YES;
+				CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = NO;
```
Any idea why this is added?

I'm not familiar with this build setting and I notice it doesn't exist in the equivalent `develop_1.x` branch.

Quick scan of Google shows up this SO thread: https://stackoverflow.com/questions/24103169/swift-compiler-error-non-modular-header-inside-framework-module

Actually... Seems like the default for this setting is already `NO`, so perhaps this change isn't even needed? _(Same for the change at line 2492)_.

rmigneco replied:
@theRealRobG Not sure why this setting was added but perhaps something that comes from building with Xcode 16? I did not deliberately add it for this work.

If it is the default, I'm content with leaving it as is if that's alright?

theRealRobG reviewed (commented):
Awesome stuff! This will finally fix #118 and fix #81 in both supported versions! 🥳 

I've left a nit-pick comment, but really out of curiosity rather than asking for any change.

I'll see if I can get a valid test for the CocoaPod approach and after that I'll approve 👍

theRealRobG reviewed (dismissed):
I tested using the following project and all looked good:
[TestMambaCocoaPods.zip](https://github.com/user-attachments/files/17692950/TestMambaCocoaPods.zip)

Some notes on how I tested:
* Used `brew install cocoapods` to install CocoaPods.
* Had to `chmod 600 ~/.netrc` for the `pod install` to work.
* Had to follow this SO answer to compile the project: https://stackoverflow.com/a/76626156
* The app imports mamba to parse and count a downloaded playlist that the user chooses via text input of a URL.

rmigneco commented:
> I tested using the following project and all looked good: [TestMambaCocoaPods.zip](https://github.com/user-attachments/files/17692950/TestMambaCocoaPods.zip)
> 
> Some notes on how I tested:
> 
> * Used `brew install cocoapods` to install CocoaPods.
> * Had to `chmod 600 ~/.netrc` for the `pod install` to work.
> * Had to follow this SO answer to compile the project: https://stackoverflow.com/a/76626156
> * The app imports mamba to parse and count a downloaded playlist that the user chooses via text input of a URL.

thank you for verifying this!

rmigneco commented:
@theRealRobG two things left here (Unless you prefer to resolve in separate PRs)

1. Providing versioning info for the SPM. We could add a text file as we do in the `develop_`.x` branch https://github.com/Comcast/mamba/blob/develop_1.x/mambaSharedFramework/Resources/version.txt
I'm happy to add this but I'm not sure if there's another preferred way. 

2. Anything you'd like to add to the README?

theRealRobG commented:
Ah, good point about the `version.txt` file @rmigneco, I think that's a good enough way of providing version information. That then needs to get updated in [FrameworkInfo.swift](https://github.com/Comcast/mamba/blob/develop/mambaSharedFramework/FrameworkInfo.swift) similar to how it was done in [develop_1.x](https://github.com/Comcast/mamba/blob/develop_1.x/mambaSharedFramework/FrameworkInfo.swift) (from memory, the version is used by mamba on writing the playlist, as it includes a comment with a Comcast copyright message as well as the library version). Without the `version.txt`, the block looking for the `CFBundleShortVersionString` fails, as with SPM the `Bundle(for: PlaylistParser.self)` doesn't exist.

rmigneco committed:
Added version.txt file for extracting framework version in SPM

- Updated Package.swift to process resources
- Updated FrameworkInfo to include the SPM path for extracting the version

rmigneco commented:
> Ah, good point about the `version.txt` file @rmigneco, I think that's a good enough way of providing version information. That then needs to get updated in [FrameworkInfo.swift](https://github.com/Comcast/mamba/blob/develop/mambaSharedFramework/FrameworkInfo.swift) similar to how it was done in [develop_1.x](https://github.com/Comcast/mamba/blob/develop_1.x/mambaSharedFramework/FrameworkInfo.swift) (from memory, the version is used by mamba on writing the playlist, as it includes a comment with a Comcast copyright message as well as the library version). Without the `version.txt`, the block looking for the `CFBundleShortVersionString` fails, as with SPM the `Bundle(for: PlaylistParser.self)` doesn't exist.

@theRealRobG this has been addressed and I think we are all set
